### ISSUE TITLE
docs: add updated log4j2 integration info

### ISSUE THREAD

**Issue by fantasticsunshine** [eclipse/jetty.project#4966]
Apache Log4j 2.10.0 introduced the class `org.apache.logging.log4j.appserver.jetty.Log4j2Logger` that can be used directly, without needing to depend on SLF4J.

There are at least 3 ways to make jetty use this new log class:

1. via system property specified on the command line:
`-Dorg.eclipse.jetty.util.log.class = org.apache.logging.log4j.appserver.jetty.Log4j2Logger`
1. via system property specified in code:
`System.setProperty("org.eclipse.jetty.util.log.class", "org.apache.logging.log4j.appserver.jetty.Log4j2Logger");`
1. by including a jetty-logging.properties file on the classpath with contents including the property:
`org.eclipse.jetty.util.log.class = org.apache.logging.log4j.appserver.jetty.Log4j2Logger`

This new class is included with the following example Maven dependency:

```
    <dependency>
      <groupId>org.apache.logging.log4j</groupId>
      <artifactId>log4j-appserver</artifactId>
      <version>${log4j.version}</version>
      <scope>runtime</scope>
    </dependency>
```

A fuller example from a Maven pom.xml looks like:

```
  <properties>
  ...
    <log4j.version>2.13.3</log4j.version>
    <jetty.version>9.4.29.v20200521</jetty.version>
  ...
  </properties>
  <dependencies>
    <dependency>
      <groupId>org.apache.logging.log4j</groupId>
      <artifactId>log4j-api</artifactId>
      <version>${log4j.version}</version>
      <scope>provided</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.logging.log4j</groupId>
      <artifactId>log4j-core</artifactId>
      <version>${log4j.version}</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.logging.log4j</groupId>
      <artifactId>log4j-appserver</artifactId>
      <version>${log4j.version}</version>
      <scope>runtime</scope>
    </dependency>
    ...
```

EDIT:
Note: log4j2.properties is still needed (to configure log4j2), in addition to jetty-logging.properties file. Those two properties files configure different things.

**Comment by joakime** [comment]
This will not work in Jetty 10 and Jetty 11.

The Jetty Logging Facade has been completely dropped and we've moved entirely over to Slf4J 2.x
Is there a Log4j 2.x that works with Slf4j 2.x yet?

**Comment by fantasticsunshine** [comment]
I've only looked as far as jetty 9.4.x, so thanks for the info about the alpha and beta versions. I just looked at SLF4J 2.x and it's only alpha too.

With that all being alpha, things may change, but it looks like Log4j 2 does provide a bridge for SLF4J 1.8 and newer: https://logging.apache.org/log4j/2.0/log4j-slf4j-impl/index.html

Specifically:
https://search.maven.org/artifact/org.apache.logging.log4j/log4j-slf4j18-impl

Log4j 2 seems pretty nice and has been shipped for a while. Is there a reason to not use it (log4j-api) directly, and instead go through an alpha release of SLF4J that may still change?

**Comment by joakime** [comment]
We've had several reports of system crashes, deadlocks, and performance issues from users of log4j2 on Jetty over the past year.

For the record, we don't think log4j2 is bad, far from it.
We think log4j2 is quite good (definitely in the top 3 logging frameworks out there), and the devs are very responsive to when we report issues to them (and pretty quickly too!, esp if they can replicate).
Right now, we are cautious of using log4j2 on high load, high contention, environments. (this cautiousness will fade over time, as it always does).

We tend to prefer logback, as its performance has consistently been the best when we test the various implementations (especially at load).
In the past 5 years, we've had 1 issue with logback (related to classloading).
Aside ... when you compare all of the logging implementations out there, java.util.logging is consistently the worst one for performance and features.

Back to why we cannot switch to something like log4j2.
Log4j2 is too specific and doesn't play nice on all of the environments Jetty runs on. (android, compact jvm profiles, osgi, alternate jvms, cloud providers, etc.).  
We chose Slf4j2 as it allows us to get out of the logging facade business and let the users choose their logging implementation. (and not surprisingly, slf4j2 works on ALL of the listed environments Jetty runs on).
Slf4j2 has released implementations (to maven central) from all logging providers (except Log4j2) right now.

**Comment by fantasticsunshine** [comment]
That's really interesting info, thanks! I think I'll look a little deeper into the newer Jetty versions and SLF4J2 and add more comments at a later date if I have something useful to add.

**Comment by fantasticsunshine** [comment]
Just following back around on the question about whether there was a log4j2.x integration with SLF4J 2.x yet. It appears after testing that the integration that I mentioned in a prior comment does work.

I was able to successfully log from Jetty 11.0.0-alpha0 to log4j 2.13.3 via SLF4J 2.0.0-alpha1 with only the following:

```
  <dependencies>
    <dependency>
      <groupId>org.eclipse.jetty</groupId>
      <artifactId>jetty-server</artifactId>
      <version>11.0.0-alpha0</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.logging.log4j</groupId>
      <artifactId>log4j-slf4j18-impl</artifactId>
      <version>2.13.3</version>
      <scope>runtime</scope>
    </dependency>
  </dependencies>
```

The `org.apache.logging.slf4j.SLF4JServiceProvider` from log4j-slf4j18-impl-2.13.3.jar is the provider that gets loaded and used. As expected, jetty-logging.properties no longer apply and all I used was a log4j2.properties file on the classpath to configure log4j2. Nice!

I'm closing this issue as it won't apply for the newer Jetty versions. I'm sure the info in this issue will be enough help for anyone on older (although currently the stable) versions.
